titus-supplementary-component/tasks-publisher/src/main/java/com/netflix/titus/supplementary/taskspublisher/EsClientHttp.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.titus.supplementary.taskspublisher;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.netflix.titus.api.json.ObjectMappers;
+import com.netflix.titus.ext.elasticsearch.TaskDocument;
+import com.netflix.titus.supplementary.taskspublisher.config.EsPublisherConfiguration;
+import io.netty.channel.ChannelOption;
+import io.netty.handler.timeout.ReadTimeoutHandler;
+import io.netty.handler.timeout.WriteTimeoutHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.tcp.TcpClient;
+
+public class EsClientHttp implements EsClient {
+    private static final Logger logger = LoggerFactory.getLogger(EsClientHttp.class);
+    private static final String ES_RECORD_TYPE = "default";
+    private final WebClient tasksClient;
+    private SimpleDateFormat indexDateFormat;
+    private EsPublisherConfiguration esPublisherConfiguration;
+
+
+    public EsClientHttp(EsPublisherConfiguration esPublisherConfiguration) {
+        this.esPublisherConfiguration = esPublisherConfiguration;
+        tasksClient = WebClient.builder().clientConnector(new ReactorClientHttpConnector(buildHttpClient()))
```

Comment:
Given the building WebClient is quite complex here, we should probably extract it into another class and inject it during construction. This way it will be easier to change the setup in the future (if need ).